apps/aggregator/src/elastic/elastic.indexer.service.ts

```diff
@@ -1,51 +1,51 @@
 import { ApiConfigService } from '@libs/common';
-import { ElasticService } from '@multiversx/sdk-nestjs-elastic';
 import { Injectable } from '@nestjs/common';
 import BigNumber from 'bignumber.js';
 import { OriginLogger } from '@multiversx/sdk-nestjs-common';
+import { ApiService, ApiSettings } from '@multiversx/sdk-nestjs-http';
+import fs from 'fs';
+import path from 'path';
 
 @Injectable()
 export class ElasticIndexerService {
   private readonly logger = new OriginLogger(ElasticIndexerService.name);
 
   constructor(
     private readonly apiConfigService: ApiConfigService,
-    private readonly elasticService: ElasticService,
+    private readonly apiService: ApiService,
   ) {}
 
   async addLiquidStakingForAddress(address: string, epoch: number, totalLiquidStaking: BigNumber): Promise<void> {
-    const currentTotalBalance = await this.getTotalBalanceWithStake(address, epoch);
-    if (currentTotalBalance.length == 0) {
-      this.logger.warn(`No entry for address ${address} found in Elasticsearch`);
+    if (!this.apiConfigService.isElasticExportEnabled()) {
       return;
     }
 
-    await this.setLiquidStakingValues(address, epoch, totalLiquidStaking, currentTotalBalance);
-  }
-
-  private async getTotalBalanceWithStake(address: string, epoch: number): Promise<string> {
-    const result = await this.elasticService.get(`${this.apiConfigService.getElasticUrl()}/${this.getIndexName(epoch)}/_search?q=_id:${address}`);
-    const hits = result.data?.hits?.hits;
-    if (hits && hits.length > 0) {
-      const totalBalanceWithStake = hits[0]._source.totalBalanceWithStake;
-      return totalBalanceWithStake == '' ? '0' : totalBalanceWithStake;
+    const indexName = this.getIndexName(epoch);
+    if (!await this.doesIndexExist(indexName)) {
+      const successful = await this.createIndexWithMapping(indexName);
+      if (!successful) {
+        return;
+      }
     }
 
-    return '';
+    await this.setLiquidStakingValues(address, epoch, totalLiquidStaking);
   }
 
-  private async setLiquidStakingValues(address: string, epoch: number, totalLiquidStaking: BigNumber, currentTotalBalance: string) {
-    let totalBalance = new BigNumber(currentTotalBalance);
-    totalBalance = totalBalance.plus(totalLiquidStaking);
-
+  private async setLiquidStakingValues(address: string, epoch: number, totalLiquidStaking: BigNumber) {
     const indexName = this.getIndexName(epoch);
-    await this.elasticService.setCustomValue(`${indexName}`, address, 'totalBalanceWithStake', totalBalance.toString(10));
-    await this.elasticService.setCustomValue(`${indexName}`, address, 'liquidStaking', totalLiquidStaking.toString(10));
+    //await this.apiService.post()
+
+    const totalLiquidStakingStr = totalLiquidStaking.toString(10);
```

Comment:
maybe we should change `toString()` to `toFixed()`. It is possible for `toString` from `bignumber.js` to stringify very large numbers in scientific notation

Reply:
done